release/mk-release-series.sh

```diff
@@ -66,8 +70,12 @@ else
   ln -s $SQUID_VCS_PATH/gitignore $SQUID_VCS_PATH/squid-$NEWVER/.git/info/exclude
 
   echo " .. Creating maintenance branches ..."
+  git checkout -b v${NEWVER}-bootstrap &&
+    git push -u origin v${NEWVER}-bootstrap
```

Comment:
> > > That branch isolates any changes from bootstrap.sh[1] automation away from other automation.
> > > [1] should only be updating documentation files now.
> > 
> > 
> > What are those "bootstrap.sh automation" (documentation) changes?

> https://github.com/squid-cache/squid/blob/master/bootstrap.sh#L154

That reference shows that squid/bootstrap.sh creates a SPONSORS file. AFAICT, that file is not _versioned_ since 2012 commit https://github.com/squid-cache/squid/commit/bdb40e3. The same bootstrap.sh script creates a lot of other unversioned files (e.g., `./configure`), of course. I see no command that add SPONSORS to some git branch. Did I miss it?


> > Perhaps you can point me to the script that does them or an already merged commit dedicated to them?

> https://github.com/squid-cache/ci/blob/main/maintenance/code-maintenance.sh#L90

AFAICT, that `runMaintenanceScript ./bootstrap.sh` command will never result in a PR because it/bootstrap.sh never modifies any git-versioned files and runMaintenanceScript() does not add any unversioned files to the branch. AFAICT, that command has no positive effect and should be removed, along with any code related to the vN-bootstrap branches (instead of adding handling for vN-bootstrap branches to this script). Am I missing something?

Reply:
This branch is protecting any changes which the bootstrap automation stage **might** be performing (now or in future). Preventing any versioned source changes from being performed then silently erased by or prior to the next step of automation.

Reply:
> This branch is protecting any changes which the bootstrap automation stage might be performing (now or in future). 

Bootstrap.sh, by design, should not perform any relevant changes. AFAICT, bootstrap.sh does not perform any relevant changes now. At the very least, no such changes are known. Bootstrap.sh will not perform any relevant changes in the future (because any bootstrap.sh modifications that add them will be blocked as buggy/misplaced).

> Preventing any versioned source changes from being performed then silently erased by or prior to the next step of automation.

If we really want to automate detecting unwanted bootstrap.sh side effects, we can do that, but creating a dedicated branch with corresponding public PRs is not the right solution even for that task.